che-incubator/che-code
# fix: environment variable is not set when running a task

### What does this PR do?
Fixes using the environment variable described in a task
```yaml
schemaVersion: 2.2.0
commands:
  - exec:
      env:
        - name: GOPATH
          value: /projects/devfile-stack-go/.go
        - name: GOCACHE
          value: /projects/devfile-stack-go/.cache
      commandLine: echo "GOCACHE=$GOCACHE" && go build main.go
```

If the environment variable is specified, the actual command that running in a terminal will have the prefix to initialize env vars
```
export GOPATH="/projects/devfile-stack-go/.go"; export GOCACHE="/projects/devfile-stack-go/.cache"; echo "GOCACHE=$GOCACHE" && go build main.go
``` 

### What issues does this PR fix?

https://github.com/eclipse-che/che/issues/22930

### How to test this PR?

- create a workspace with https://github.com/vitaliy-guliy/devfile-stack-go/tree/test-tasks?editor-image=quay.io/che-incubator-pull-requests/che-code:pr-367-amd64
- run `devfile:build` command
- ensure `main` binary file is created
- check the terminal, ensure `GOCACHE` environment variable is set
- you can run `devfile:test` command to check the behavior when the variable value contains quotes

### Does this PR contain changes that override default upstream Code-OSS behavior?
